server/src/main/java/io/deephaven/server/table/ops/ApplyPreviewColumnsGrpcImpl.java

```diff
@@ -17,14 +18,16 @@
 @Singleton
 public class ApplyPreviewColumnsGrpcImpl extends GrpcTableOperation<ApplyPreviewColumnsRequest> {
     @Inject
-    protected ApplyPreviewColumnsGrpcImpl() {
-        super(BatchTableRequest.Operation::getApplyPreviewColumns, ApplyPreviewColumnsRequest::getResultId,
-                ApplyPreviewColumnsRequest::getSourceId);
+    protected ApplyPreviewColumnsGrpcImpl(final TableServiceContextualAuthWiring authWiring) {
+        super(authWiring::checkPermissionApplyPreviewColumns, BatchTableRequest.Operation::getApplyPreviewColumns,
+                ApplyPreviewColumnsRequest::getResultId, ApplyPreviewColumnsRequest::getSourceId);
     }
 
     @Override
-    public Table create(ApplyPreviewColumnsRequest request, List<SessionState.ExportObject<Table>> sourceTables) {
+    public Table create(final ApplyPreviewColumnsRequest request,
+            final List<SessionState.ExportObject<Table>> sourceTables) {
         Assert.eq(sourceTables.size(), "sourceTables.size()", 1);
+        checkPermission(request, sourceTables);
         return ColumnPreviewManager.applyPreview(sourceTables.get(0).get());
     }
```

Comment:
I know I mentioned abstracting checkPermission - much better now IMO.

Does it make sense to take it a step further? Have GrpcTableOperation implement create(), and introduce a new abstract method `createImpl` where the table concrete GrpcTableOperation implementation doesn't even have to call checkPermission themselves?

Reply:
I lifted the checkPermission call to the callers of create. There are only two, and they are controlled by the batching / one-shot RPC code paths. I like this better than having a `create` and `createImpl`.